plugins/resources-generator/src/main/kotlin/dev/icerock/gradle/generator/jvm/JvmPluralsGenerator.kt

```diff
@@ -0,0 +1,86 @@
+package dev.icerock.gradle.generator.jvm
+
+import com.squareup.kotlinpoet.CodeBlock
+import com.squareup.kotlinpoet.KModifier
+import dev.icerock.gradle.generator.KeyType
+import dev.icerock.gradle.generator.PluralMap
+import dev.icerock.gradle.generator.PluralsGenerator
+import dev.icerock.gradle.generator.jvm.JvmStringsGenerator.Companion.replaceAndroidFormatParameters
+import org.gradle.api.file.FileTree
+import java.io.File
+
+class JvmPluralsGenerator(pluralsFileTree: FileTree) : PluralsGenerator(pluralsFileTree) {
+
+    override fun getClassModifiers(): Array<KModifier> = arrayOf(KModifier.ACTUAL)
+
+    override fun getPropertyModifiers(): Array<KModifier> = arrayOf(KModifier.ACTUAL)
+
+    override fun getPropertyInitializer(key: String, baseLanguageMap: Map<KeyType, PluralMap>) =
+        CodeBlock.of(
+            "PluralsResource(%S, %L)",
+            key,
+            getNumberFormat(key, baseLanguageMap)
+        )
+
+    override fun generateResources(
+        resourcesGenerationDir: File,
+        language: String?,
+        strings: Map<KeyType, PluralMap>
+    ) {
+        val fileDirName = when (language) {
+            null -> PLURALS_BUNDLE_NAME
+            else -> "${PLURALS_BUNDLE_NAME}_$language"
+        }
+
+        val localizationDir = File(resourcesGenerationDir, LOCALIZATION_DIR).apply { mkdirs() }
+        val stringsFile = File(localizationDir, "${fileDirName}.properties")
+
+        val content = strings.map { (key, pluralMap) ->
+            "$key = {0}\n" +
+                    pluralMap.map { (quantity, value) ->
+                        "${
+                            getQuantityKey(
+                                key = key,
+                                quantity = quantity
+                            )
+                        } = ${value.replaceAndroidFormatParameters()}"
+                    }.joinToString("\n")
+        }.joinToString("\n")
+
+        stringsFile.writeText(content)
+    }
+
+    private fun getNumberFormat(key: String, baseLanguageMap: Map<KeyType, PluralMap>) =
+        baseLanguageMap[key]?.let {
+            "listOf(${
+                it.mapNotNull { (quantity, _) ->
+                    val quantityInNumber =
+                        mapAndroidQuantityToDouble(quantity) ?: return@mapNotNull null
+
+                    "${quantityInNumber.toDouble()} to \"${
+                        getQuantityKey(
+                            key = key,
+                            quantity = quantity
+                        )
+                    }\""
+                }.joinToString()
+            })"
```

Comment:
it should move into moko-resources runtime library (support autogenerated code difficult)

Reply:
@Alex009 I'm not sure I follow you on this, could you provide an example?

Reply:
i check generated code and see that we should not move code to lib, but we should simplify this generator code logic. move code away from "${}"